docs/source/tutorials/formatting_eureka_s4_compatible_data.ipynb

```diff
@@ -0,0 +1,588 @@
+    "* The error on the extracted spectrum at each time step (`opterr`), and units.\n",
+    "\n",
+    "**Optional**\n",
+    "\n",
+    "* A 2D mask for specific points to ignore in the spectral time series (`optmask`).\n",
+    "* The pixel shift in the peak of the spectral trace over time (`centroid_y` / `centroid_sy`).\n",
+    "* Various other variables and attributes.\n",
+    "&nbsp;\n",
+    "\n",
+    "Using this as a rough guide, let's build a mock dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "f8357a19-f96f-40ad-959b-53aec69bd53d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "\n",
+    "time = np.linspace(59694.0, 59694.1, 100)\n",
+    "time_units = 'MJD'\n",
+    "wave_1d = np.linspace(1.0, 5.0, 1001)\n",
+    "x = np.arange(len(wave_1d))\n",
+    "wave_units = 'microns'\n",
+    "optspec = np.ones((len(time), len(wave_1d)))\n",
+    "opterr = np.ones((len(time), len(wave_1d)))*0.01\n",
+    "flux_units = 'e-'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6e1573b1-6745-4b23-86f0-740d6e95d00a",
+   "metadata": {},
+   "source": [
+    "Note that these values are entirely made up for the purpose of this notebook - you should be setting these variables *your* data values!\n",
+    "&nbsp;\n",
+    "\n",
+    "To build the xarray, we will again use the [astraeus](https://kevin218.github.io/Astraeus/) library. Let's start by initialising a dataset:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "e091de62-12b0-4f56-b29f-88dfd52b6014",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from astraeus.xarrayIO import makeDataset\n",
+    "\n",
+    "custom_dataset = makeDataset()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "13520c0f-d3ab-4b31-a94d-1e4e960d3b45",
+   "metadata": {},
+   "source": [
+    "Now we can assign all of the arrays from above to the correct Xarray DataArray format: "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "a0b8a690-431b-401b-ba82-e3fbf5cb75af",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "custom_dataset['time'] = time\n",
+    "custom_dataset['time'].attrs['time_units'] = time_units\n",
+    "custom_dataset['x'] = x\n",
+    "custom_dataset['optspec'] = (['time', 'x'], optspec)\n",
+    "custom_dataset['optspec'].attrs['flux_units'] = flux_units\n",
+    "custom_dataset['optspec'].attrs['time_units'] = time_units\n",
+    "custom_dataset['optspec'].attrs['wave_units'] = wave_units\n",
+    "custom_dataset['opterr'] = (['time', 'x'], opterr)\n",
+    "custom_dataset['opterr'].attrs['flux_units'] = flux_units\n",
+    "custom_dataset['opterr'].attrs['time_units'] = time_units\n",
+    "custom_dataset['opterr'].attrs['wave_units'] = wave_units\n",
+    "custom_dataset['wave_1d'] = (['x'], wave_1d)\n",
+    "custom_dataset['wave_1d'].attrs['wave_units'] = wave_units"
```

Comment:
I had issues with this when I coding things up in January, and found that being explicit with the generation of the array was easiest. Feel free to try and tweak things!

Reply:
@kevin218, I just tried using Astraeus' built-in functions to do this, and there is indeed an issue with doing so. The SpecData files should have the format `optspec = (time, x)`, but using Astraeus' `makeLCDA` function on the `optspec` array I get the transpose: `(wavelength, time)`. In addition to the transpose issue, I'm also not totally sure whether having `wavelength` instead of `x` would break things within Eureka!'s Stage 4. Using the `makeFluxLikeDA` function also doesn't work since it requires that the input array be three dimensional `(time, x, y)` while we only have/want `(time, x)`

Reply:
Ok, that's too bad. In an ideal world, I could create a function quickly, but that won't happen until after Exo5 and vacation.

Can you add an issue in Astraeus?

Reply:
I've opened an issue there and have linked to this comment so you'll have the reminder of what the issue was